components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -1354,6 +1339,49 @@ private static String getTenantUrl(String url, String tenantDomain) throws URISy
         }
     }
 
+    /**
+     * Builds a URL with a given context in both the tenant-qualified url supported mode and the legacy mode. Returns
+     * the absolute URL build from the default context in the tenant-qualified url supported mode and returns the
+     * absolute url build from file configuration context, proxy context path and web context path.
+     *
+     * @param defaultContext                  Default URL context.
+     * @param getValueFromFileBasedConfig     File-based Configuration.
+     * @param addProxyContextPathInLegacyMode Add proxy context path to the URL.
+     * @param addWebContextRootInLegacyMode   Add web context path to the URL.
+     * @return Absolute URL.
+     */
+    public static String buildUrl(String defaultContext, Supplier<String> getValueFromFileBasedConfig,
+                                  boolean addProxyContextPathInLegacyMode, boolean addWebContextRootInLegacyMode) {
+
+        String url;
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            url = buildTenantQualifiedUrl(defaultContext);
+        } else {
+            String oauth2EndpointURLInFile = null;
+            if (getValueFromFileBasedConfig != null) {
+                oauth2EndpointURLInFile = getValueFromFileBasedConfig.get();
+            }
+            if (StringUtils.isNotBlank(oauth2EndpointURLInFile)) {
+                // Use the value configured in the file.
+                url = oauth2EndpointURLInFile;
+            } else {
+                // Use the default context.
+                url = IdentityUtil
+                        .getServerURL(defaultContext, addProxyContextPathInLegacyMode, addWebContextRootInLegacyMode);
```

Comment:
This API should be deprecated and not to be used any more. Only proxy context path will be utilized from new ServiceURLBuilder API.